### PR TITLE
refactor pseudopotentials

### PR DESCRIPTION
**Pull request by mmezic**
Changes:
- Most of the code related to pseudopotentials has been moved to a new folder called `pp`.
- When using pseudopotentials, they are now turned off by default for H and He. This can be overridden by specifying `pp_mask`.
- Optimized performance when using PPs for H and He. It now avoids computing terms that would be multiplied by zero.
- All pseudopotential information is now stored in the hamiltonian instead of the molecule. Hence some important variables, e.g. number of (up/down) electrons, have also been moved to the hamiltonian. Arguments of some methods therefore had to be changed from `mol` -> `hamil`.
- Removed direct execution of a jitted function from the top-level scope. `import deepqmc` thus no longer causes an error when jax is installed incorrectly.
- The nuclear potential and pseudopotentials are now defined as classes called `NuclearCoulombPotential` and `EcpTypePseudopotential`. The letter includes both `ccECP` and `bfd` pseudopotentials.
- Both documentation and tests have also been updated to reflect these changes.

### CODE REVIEW

File: src/deepqmc/gnn/electron_gnn.py [hunk 1]
```diff
@@ -332,10 +332,10 @@ def __init__(
         ghost_coords=None,
     ):
         super().__init__()
-        n_nuc, n_up, n_down = mol.n_particles
+        n_nuc, n_up, n_down = hamil.mol.n_nuc, hamil.n_up, hamil.n_down
```
Comment (szbernat) [inline src/deepqmc/gnn/electron_gnn.py:335]: Now it's a little odd that `n_up` and `n_down` is on the Hamiltonian, but `n_nuc` is on the Molecule. Maybe move everything to the Hamiltonian?

File: src/deepqmc/wf/baseline/pyscfext.py [hunk 2]
```diff
@@ -5,12 +5,10 @@
 from pyscf.mcscf import CASSCF
 from pyscf.scf import RHF
 
-from ...utils import pad_list_of_3D_arrays_to_one_array
-
 log = logging.getLogger(__name__)
 
 
-def pyscf_from_mol(mol, basis, cas=None, **kwargs):
+def pyscf_from_mol(hamil, basis, cas=None, **kwargs):
```
Comment (szbernat) [inline src/deepqmc/wf/baseline/pyscfext.py:11]: We should update the name and docstring of this function `mol`->`hamil`.

**Review by szbernat — COMMENTED** [review]
I like it!

File: src/deepqmc/gnn/electron_gnn.py (hunk 1 above)
Comment (mmezic) [reply src/deepqmc/gnn/electron_gnn.py]: I agree, I've moved `n_nuc` to Hamiltonian in the last commit

File: src/deepqmc/wf/baseline/pyscfext.py (hunk 2 above)
Comment (mmezic) [reply src/deepqmc/wf/baseline/pyscfext.py:11]: Good point, I've updated the docstring

**Review by szbernat — APPROVED** [review]

**Comment by szbernat** [comment]
Ok, now comes the million dollar question of rebasing this onto the recently merged stuff. In principle the changes here should be relatively orthogonal, but I'm sure that there'll be some conflicts.

**Comment by mmezic** [comment]
There were some collisions with the multi-GPU implementation, but I rebased it and it seems to work. Can we merge it to master?

**Comment by szbernat** [comment]
Sounds great!

What kind of tests did you do?

Maybe we could check that the following jobs give reasonable energies:
1. Actual multi GPU calculation (perhaps on allegro) without pseudopotentials. Something simple like LiH.
2. Actual multi GPU calculation with pseudopotentials. Not sure what's a good small testcase here.

**Comment by mmezic** [comment]
I've done some multi GPU tests on various systems with and without pseudopotentials and (after fixing a small bug in `paulinet` ansatz) the energies are correct, so I think we're good to go.

**Pull request merged**